libtsuba/src/tsuba.cpp

```diff
@@ -232,6 +233,157 @@ tsuba::ListAvailableViews(const std::string& rdg_dir) {
+      // We're batching this now because we want to rely on having the RDG manifest file in case things
+      // change in the future.
+      katana::Uri dst_file_uri;
+      if (tsuba::RDGManifest::IsPartitionFileUri(src_file_uri)) {
+        KATANA_LOG_WARN("src_file_uri partition: {}", src_file_uri);
+        auto host_id =
+            KATANA_CHECKED(tsuba::RDGManifest::ParseHostFromPartitionFile(
+                src_file_uri.BaseName()));
+        auto dst_dir_uri = KATANA_CHECKED(katana::Uri::Make(dst_dir));
+        dst_file_uri = tsuba::RDGManifest::PartitionFileName(
+            rdg_manifest_res.value().view_type(), dst_dir_uri, host_id, 1);
+        KATANA_LOG_WARN("dst_file_uri partition: {}", dst_file_uri);
+      } else {
+        auto dst_file_path = katana::Uri::JoinPath(dst_dir, fname);
+        dst_file_uri = KATANA_CHECKED(katana::Uri::Make(dst_file_path));
+        KATANA_LOG_WARN("src_file_uri: {}", src_file_uri);
+        KATANA_LOG_WARN("dst_file_uri: {}", dst_file_uri);
+      }
+
+      src_dst_pairs.push_back(std::make_pair(src_file_uri, dst_file_uri));
+    }
+
+    // We add the manifest file to the vector
+    // Set the version to be 1
+    auto rdg_manifest_uri = rdg_manifest_res.value().FileName();
+    rdg_manifest_res.value().ResetVersion();
+    auto dst_rdg_manifest_path = katana::Uri::JoinPath(
+        dst_dir, rdg_manifest_res.value().FileName().BaseName());
+    auto dst_rdg_manifest_uri =
+        KATANA_CHECKED(katana::Uri::Make(dst_rdg_manifest_path));
+    KATANA_LOG_WARN("rdg_manifest_uri: {}", rdg_manifest_uri);
+    KATANA_LOG_WARN("dst_rdg_manifest_uri: {}", dst_rdg_manifest_uri);
+    src_dst_pairs.push_back(
+        std::make_pair(rdg_manifest_uri, dst_rdg_manifest_uri));
+  }
+  return src_dst_pairs;
+}
+
+katana::Result<void>
+tsuba::CopyRDG(std::vector<std::pair<katana::Uri, katana::Uri>> src_dst_pairs) {
+  // TODO: make sure that manifests are written at the end!
+  // TODO: add do_all loop
+  std::vector<uint64_t> manifest_uri_idxs;
+  for (uint64_t i = 0; i < src_dst_pairs.size(); i++) {
+    auto [src_file_uri, dst_file_uri] = src_dst_pairs[i];
+    // We save the names of all the manifest files and we write them out at the end.
+    if (tsuba::RDGManifest::IsManifestUri(src_file_uri)) {
+      manifest_uri_idxs.push_back(i);
+      continue;
+    }
+    tsuba::FileView fv;
+    KATANA_CHECKED(fv.Bind(src_file_uri.path(), false));
+    KATANA_CHECKED(
+        tsuba::FileStore(dst_file_uri.path(), fv.ptr<char>(), fv.size()));
+  }
+
+  // Process all the manifest files, write them out.
+  // We want to write this last so that we know whether a write fully finished or not.
+  for (auto idx : manifest_uri_idxs) {
+    auto [src_file_uri, dst_file_uri] = src_dst_pairs[idx];
+    auto rdg_manifest = KATANA_CHECKED(tsuba::RDGManifest::Make(src_file_uri));
+    // These are hard-coded for now. Will what we copy always be version 1?
+    // Should we clear the lineage as well?
```

Comment:
Should reset the lineage of the first version, but others are fine to keep I'd think.

Reply:
If we have multiple views, we will be copying all of them (since they would all have the same version) and then setting their versions to 1. Does this mean we reset the lineages for all of them?

Reply:
The lineage has never been used for anything and my guess is that it never will, but the idea was to give users some idea of how version N was created from version N-1.  I'd think every view should have the same lineage, because all views are derived from version N-1 the same way.

Reply:
I didn't get a clear consensus on this. Should I just clear out any lineage information by default? I was thinking that lineage information would be useful for when we want to backtrack changes or for fault tolerance/corruption?

Reply:
The log is for fault tolerance.  Lineage is just documentation.  If you are copying every file, copy the lineage.

Reply:
So keep the lineage as is?

Reply:
Yes.